File: 70-Climbing-Stairs.py
class Solution:
    def climbStairs(self, n: int) -> int:

        def solve(i, dp):

            if i<0:
                return 
            
            if i==0 or i==1 or i==2:
                return i

            if dp[i] !=-1:
                return dp[i]

            step1 = solve(i-1, dp)
            step2 = solve(i-2, dp)

            dp[i] = step1 +step2
            return dp[i]

        dp = [-1] * (n+1)
        return solve(n, dp)

            

File: 70-Climbing-Stairs.cpp
class Solution {
public:
    int climbStairs(int n) {
        vector<int> arr(n+1);
        if(n==0|| n==1|| n==2|| n==3){
            return n;
        }
        arr[0]=0;
        arr[1] =1;
        arr[2] = 2;

        for(int i=3;i<=n;i++){
            arr[i] = arr[i-1]+arr[i-2];
        }
        return arr[n];
    }
};